core/src/main/java/org/testcontainers/utility/TestcontainersConfiguration.java

```diff
@@ -101,9 +101,22 @@ public boolean isDisableChecks() {
 
     @UnstableAPI
     public boolean environmentSupportsReuse() {
-        return Boolean.parseBoolean((String) environmentProperties.getOrDefault("testcontainers.reuse.enable", "false"));
-    }
-
+        return Boolean.parseBoolean((String) environmentProperties.getOrDefault("testcontainers.reuse.enable", "false"))
+                || !isCI();
+    }
+    
+    private boolean isCI() {
```

Comment:
I am afraid we cannot use this approach. While it does support some percentage of environments, it is too easy to break it:
```bash
$ docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock maven:3 ./mvnw test
```

The feature is powerful but dangerous, this is why we decided to make it opt-in, not opt-out.

Reply:
pardon my ignorance but what do you mean by saying `it is too easy to break it <docker cmd>`? From looking at it I'm not sure why that command would be an issue. Is it because if a developer runs a maven build inside a docker container locally the CI env vars wouldn't be set and therefore containers would get leftover?

Reply:
it is a common way of running tests inside a container on CIs.
Docker won't pass host's environment variables, leading to reuse being activated.

Reply:
ok, thanks for explaining that. I didn't realize that was a common pattern (docker-in-docker-in-docker) but it would certainly be an issue. I'll go ahead and close out this PR since the approach won't work.

Reply:
@aguibert maybe you can keep it for the `log.warn`? It still makes sense to improve that message 👍

Reply:
ah, I had completely forgotten about that! Let me re-open and include just that change

Reply:
opened a separate PR here: https://github.com/testcontainers/testcontainers-java/pull/2422